### PR TITLE
Add testing helper for consistent channel/mixture

### LINKED ISSUE #2271: Add validation for Kraus operators
Many of the methods involving noise/channels mention that the matrices making up a mixture or channel must satisfy \sum_{k=0}^{r-1} A_i^\dagger A_i = I. This is never enforced, which allows the user to simulate unphysical systems and get invalid density matrices (see example below).

Validation is easy enough (`assert np.sum([m.T @ m for m in object._channel_()`) but I'm not sure where to put this @Strilanc ? Ideally this will be called exactly once, before a call to `simulate`. 

Code Example:
```
from typing import Iterable

import cirq
import numpy as np

class UnnormalizedChannel(cirq.GeneralizedAmplitudeDampingChannel):
    """Overwrite the correct amplitude damping channel with a bad one."""
    def _channel_(self) -> Iterable[np.ndarray]:
        p0 = 1
        sqrt_g = np.sqrt(self._gamma)
        return (
            p0 * np.array([[1., 0.], [0., 1.]]),
            p0 * np.array([[0., sqrt_g], [0., 0.]]),
        )


def unnormalized_channel(gamma):
    return UnnormalizedChannel(1.0, gamma)


q0 = cirq.LineQubit(0)
bad_noise_inst = unnormalized_channel(0.3)
c = cirq.Circuit.from_ops(cirq.X(q0), bad_noise_inst.on(q0))
dm = cirq.DensityMatrixSimulator().simulate(c).final_density_matrix
kraus_sum = np.sum(m.T @ m for m in bad_noise_inst._channel_())
print(np.allclose(kraus_sum, np.eye(2)))
print(np.isclose(np.trace(dm), 1.0))
```
which prints
```
>>> False
>>> False
```

### PR DESCRIPTION
**Pull request by dabacon**
Adds a helper that tests that channel kraus operator is cptp (\sum_i A_i^\dagger A_i = I) and that has_kraus is true.  Uses this on channels.

Adds a helper for testing that mixture's probabilities sum to 1.  Uses this on mixtures.

Fixes #2271

### CODE REVIEW

**Review by MichaelBroughton — COMMENTED** [review]
Could we also add similar for _mixture_ ?

**Comment by 95-martin-orion** [comment]
Related: https://github.com/quantumlib/Cirq/pull/4365

File: cirq-core/cirq/testing/consistent_channel.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import cirq
+
+
+def assert_consistent_channel(gate: Any, rtol: float = 1e-5, atol: float = 1e-8):
+    assert cirq.has_kraus(gate), f"Given gate {gate!r} does not return True cirq.has_kraus."
+    kraus_ops = cirq.kraus(gate)
+    assert cirq.is_cptp(kraus_ops=kraus_ops, rtol=rtol, atol=atol), (
+        f"Kraus operators for {gate!r} did not sum to identity up to expected tolerances. "
+        f"Summed to {sum(m.T.conj() @ m for m in kraus_ops)}"
+    )
```
Comment (tanujkhattar) [inline cirq-core/cirq/testing/consistent_channel.py:26]: Please also add tests for has_mixture and mixture.

Comment (dabacon) [reply cirq-core/cirq/testing/consistent_channel.py:26]: this is now done

**Review by MichaelBroughton — APPROVED** [review]
LGTM

**Pull request merged**